### PR TITLE
Make tye exit if dotnet fails to start projects

### PR DESCRIPTION
**Pull request by jkotalik**
Fixes https://github.com/dotnet/tye/issues/543.

image

### CODE REVIEW

File: src/Microsoft.Tye.Hosting/TyeBuildException.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+        public TyeBuildException(string? message) : base(message)
+        {
+        }
+
+        public TyeBuildException(string? message, Exception? innerException) : base(message, innerException)
+        {
+        }
+
+        protected TyeBuildException(SerializationInfo info, StreamingContext context) : base(info, context)
```
Comment (JunTaoLuo) [inline src/Microsoft.Tye.Hosting/TyeBuildException.cs:25]: Nit: Not all of these constructors are needed right?

File: src/Microsoft.Tye.Hosting/TyeBuildException.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.Serialization;
+
+namespace Microsoft.Tye.Hosting
+{
+    [Serializable]
```
Comment (JunTaoLuo) [inline src/Microsoft.Tye.Hosting/TyeBuildException.cs:10]: nit: is this required?

File: src/Microsoft.Tye.Hosting/TyeHost.cs [hunk 3]
```diff
@@ -257,7 +265,7 @@ private static bool IsPortInUseByBinding(Application application, int port)
             return false;
         }
 
-        private static AggregateApplicationProcessor CreateApplicationProcessor(ReplicaRegistry replicaRegistry, HostOptions options, Microsoft.Extensions.Logging.ILogger logger)
+        private static AggregateApplicationProcessor CreateApplicationProcessor(ReplicaRegistry replicaRegistry, HostOptions options, Microsoft.Extensions.Logging.ILogger logger, IHostApplicationLifetime lifetime)
```
Comment (JunTaoLuo) [inline src/Microsoft.Tye.Hosting/TyeHost.cs:268]: nit: this change isn't needed anymore?

File: src/Microsoft.Tye.Hosting/ProcessRunner.cs [hunk 4]
```diff
@@ -11,8 +11,10 @@
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Build.Globbing.Extensions;
```
Comment (JunTaoLuo) [inline src/Microsoft.Tye.Hosting/ProcessRunner.cs:14]: I'm guessing some of these imports are no longer used?

File: src/Microsoft.Tye.Hosting/TyeBuildException.cs (hunk 1 above)
Comment (jkotalik) [reply src/Microsoft.Tye.Hosting/TyeBuildException.cs:25]: These are autogenerated when creating an exception type, I'd rather not modify it. IIRC it's a requirement for exceptions in the runtime repo to have these constructors as well.

**Pull request merged**